src/benchmarks/micro/MicroBenchmarks.csproj

```diff
@@ -14,9 +14,8 @@
     <DebugType>portable</DebugType>
     <DebugSymbols>true</DebugSymbols>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <EnableXlfLocalization>false</EnableXlfLocalization>
-    <!-- this repo needs to build with netcoreapp3.1, so we can't use latest C# syntax -->
-    <LangVersion>8.0</LangVersion>
+    <EnableXlfLocalization>false</EnableXlfLocalization>    
+    <LangVersion>9.0</LangVersion>
```

Comment:
> FWIW I have checked locally and it compiles for all TFMs.

Which SDK did you use? The problem with 3.1 is that if you use newer SDK (6.0+) it builds fine when it targets quite recent LangVersion. But when you use 3.1 SDK, it fails.

To test that we need to use the python scripts:

```cmd
py .\scripts\benchmarks_ci.py -f netcoreapp3.1
```

Edit: I've tested your fork and everything builds fine for 3.1. 👍